Core/Core/Submissions/GetSubmissions.swift

```diff
@@ -151,3 +151,106 @@ public class GetSubmission: APIUseCase {
+    }
+
+    public var scope: Scope {
+        var predicate = NSPredicate(key: #keyPath(Submission.assignmentID), equals: assignmentID)
+        if let filter = filter?.predicate {
+            predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [ predicate, filter ])
+        }
+        let order = shuffled ?
+            NSSortDescriptor(key: #keyPath(Submission.shuffleOrder), ascending: true) :
+            NSSortDescriptor(key: #keyPath(Submission.sortableName), naturally: true)
+        return Scope(predicate: predicate, order: [order])
+    }
+
+    public enum Filter: RawRepresentable {
+        case late, notSubmitted, needsGrading, graded
+        case scoreAbove(Double)
+        case scoreBelow(Double)
+
+        public init?(rawValue: String?) {
+            guard let rawValue = rawValue else { return nil }
+            self.init(rawValue: rawValue)
+        }
+
+        public init?(rawValue: String) {
+            switch rawValue {
+            case "late":
+                self = .late
+            case "not_submitted":
+                self = .notSubmitted
+            case "needs_grading":
+                self = .needsGrading
+            case "graded":
+                self = .graded
+            default:
+                let parts = rawValue.split(separator: "_")
+                if parts.count == 3, parts[0] == "score", parts[1] == "above", let score = Double(parts[2]) {
+                    self = .scoreAbove(score)
+                } else if parts.count == 3, parts[0] == "score", parts[1] == "above", let score = Double(parts[2]) {
+                    self = .scoreAbove(score)
+                } else {
+                    return nil
+                }
+            }
+        }
+
+        public var rawValue: String {
+            switch self {
+            case .late:
+                return "late"
+            case .notSubmitted:
+                return "not_submitted"
+            case .needsGrading:
+                return "needs_grading"
+            case .graded:
+                return "graded"
+            case .scoreAbove(let score):
+                return "score_above_\(score)"
+            case .scoreBelow(let score):
+                return "score_below_\(score)"
+            }
+        }
+
+        var predicate: NSPredicate {
+            switch self {
+            case .late:
+                return NSPredicate(key: #keyPath(Submission.late), equals: true)
+            case .notSubmitted:
+                return NSPredicate(key: #keyPath(Submission.submittedAt), equals: nil)
+            case .needsGrading:
+                return NSPredicate(format: "%K != nil AND %K == nil",
+                    #keyPath(Submission.submittedAt),
+                    #keyPath(Submission.gradedAt)
+                )
+            case .graded:
+                return NSPredicate(format: "%K != nil", #keyPath(Submission.gradedAt))
```

Comment:
This would also have to check `excused`, I believe. See `Submission.isGraded`. But if this is how RN currently works I won't worry so much about it.